tests/test_stateful_browser.py

```diff
@@ -295,6 +295,10 @@ def test_form_multiple():
     response = browser.submit_selected()
     browser.close()
 
+def test_with():
```

Comment:
I know it's not the job of documentation to teach people how python works, but I wouldn't have know that `with` calls `__enter__` and `__exit__` without looking it up. Maybe we could add a comment in the `test_with` that says that "testing with" <==> "testing enter and exit"? Certainly not necessary though!

Reply:
done.